*** text/draft-muffett-end-to-end-secure-messaging-00.txt	2021-05-07 22:51:10.000000000 +0100
--- text/draft-muffett-end-to-end-secure-messaging-01.txt	2021-05-10 20:00:16.000000000 +0100
***************
*** 4,15 ****
  
  individual submission                                         A. Muffett
  Internet-Draft                                       Security Researcher
! Intended status: Informational                                7 May 2021
! Expires: 8 November 2021
  
  
            Functional Definition of End-to-End Secure Messaging
!               draft-muffett-end-to-end-secure-messaging-00
  
  Abstract
  
--- 4,15 ----
  
  individual submission                                         A. Muffett
  Internet-Draft                                       Security Researcher
! Intended status: Informational                               10 May 2021
! Expires: 11 November 2021
  
  
            Functional Definition of End-to-End Secure Messaging
!               draft-muffett-end-to-end-secure-messaging-01
  
  Abstract
  
***************
*** 33,39 ****
     time.  It is inappropriate to use Internet-Drafts as reference
     material or to cite them other than as "work in progress."
  
!    This Internet-Draft will expire on 8 November 2021.
  
  Copyright Notice
  
--- 33,39 ----
     time.  It is inappropriate to use Internet-Drafts as reference
     material or to cite them other than as "work in progress."
  
!    This Internet-Draft will expire on 11 November 2021.
  
  Copyright Notice
  
***************
*** 53,59 ****
  
  
  
! Muffett                  Expires 8 November 2021                [Page 1]
  
  Internet-Draft                    e2esm                         May 2021
  
--- 53,59 ----
  
  
  
! Muffett                 Expires 11 November 2021                [Page 1]
  
  Internet-Draft                    e2esm                         May 2021
  
***************
*** 79,88 ****
       4.2.  Conversation  . . . . . . . . . . . . . . . . . . . . . .   5
       4.3.  Plaintext Content and Sensitive Metadata (PCASM)  . . . .   5
         4.3.1.  Content PCASM . . . . . . . . . . . . . . . . . . . .   5
!        4.3.2.  Size PCASM  . . . . . . . . . . . . . . . . . . . . .   5
!        4.3.3.  Descriptive PCASM . . . . . . . . . . . . . . . . . .   6
         4.3.4.  Conversation Metadata (OPTIONAL)  . . . . . . . . . .   6
!        4.3.5.  Non-PCASM . . . . . . . . . . . . . . . . . . . . . .   6
       4.4.  Backdoor  . . . . . . . . . . . . . . . . . . . . . . . .   7
         4.4.1.  Why call this a "backdoor"? . . . . . . . . . . . . .   7
     5.  Scope of a Participant in E2ESM . . . . . . . . . . . . . . .   8
--- 79,88 ----
       4.2.  Conversation  . . . . . . . . . . . . . . . . . . . . . .   5
       4.3.  Plaintext Content and Sensitive Metadata (PCASM)  . . . .   5
         4.3.1.  Content PCASM . . . . . . . . . . . . . . . . . . . .   5
!        4.3.2.  Size PCASM  . . . . . . . . . . . . . . . . . . . . .   6
!        4.3.3.  Analytic PCASM  . . . . . . . . . . . . . . . . . . .   6
         4.3.4.  Conversation Metadata (OPTIONAL)  . . . . . . . . . .   6
!        4.3.5.  Non-PCASM . . . . . . . . . . . . . . . . . . . . . .   7
       4.4.  Backdoor  . . . . . . . . . . . . . . . . . . . . . . . .   7
         4.4.1.  Why call this a "backdoor"? . . . . . . . . . . . . .   7
     5.  Scope of a Participant in E2ESM . . . . . . . . . . . . . . .   8
***************
*** 109,115 ****
  
  
  
! Muffett                  Expires 8 November 2021                [Page 2]
  
  Internet-Draft                    e2esm                         May 2021
  
--- 109,115 ----
  
  
  
! Muffett                 Expires 11 November 2021                [Page 2]
  
  Internet-Draft                    e2esm                         May 2021
  
***************
*** 141,147 ****
     Software which functions as an End-to-End Secure Messenger MUST
     satisfy the following principles, and MUST satisfy these principles
     in respect of the provided definitions for all forms of communication
!    and data-sharing that the software offers.
  
     Any software that does not satisfy these requirements is not an End-
     to-End Secure Messenger, and it does not implement End-to-End Secure
--- 141,149 ----
     Software which functions as an End-to-End Secure Messenger MUST
     satisfy the following principles, and MUST satisfy these principles
     in respect of the provided definitions for all forms of communication
!    and data-sharing that the software offers.  The software MAY comprise
!    either a complete application, or a clearly defined subset of
!    functionality within a larger application.
  
     Any software that does not satisfy these requirements is not an End-
     to-End Secure Messenger, and it does not implement End-to-End Secure
***************
*** 163,171 ****
  
  
  
! 
! 
! Muffett                  Expires 8 November 2021                [Page 3]
  
  Internet-Draft                    e2esm                         May 2021
  
--- 165,171 ----
  
  
  
! Muffett                 Expires 11 November 2021                [Page 3]
  
  Internet-Draft                    e2esm                         May 2021
  
***************
*** 211,231 ****
  
  3.5.  Management and Visibility of Participant Clients and Devices
  
!    E2ESM software MUST provide each participant entity with means to
!    review or revoke access for clients or devices that can access future
!    PCASM.
  
!    E2ESM software MUST provide each participant entity with
!    notifications and/or complete logs of changes to the set of clients
!    or devices that can or could access message PCASM.
  
  
  
! Muffett                  Expires 8 November 2021                [Page 4]
  
  Internet-Draft                    e2esm                         May 2021
  
  
  4.  Definitions
  
     These principles MUST be measured with respect to the following
--- 211,235 ----
  
  3.5.  Management and Visibility of Participant Clients and Devices
  
!    Where there exists centralised E2ESM software that hosts
!    participants:
  
!    1.  The E2ESM software MUST provide each participant entity with
!        means to review or revoke access for clients or devices that can
!        access future PCASM.
  
  
  
! 
! Muffett                 Expires 11 November 2021                [Page 4]
  
  Internet-Draft                    e2esm                         May 2021
  
  
+    2.  The E2ESM software MUST provide each participant entity with
+        notifications and/or complete logs of changes to the set of
+        clients or devices that can or could access message PCASM.
+ 
  4.  Definitions
  
     These principles MUST be measured with respect to the following
***************
*** 235,241 ****
  
     A participant is any entity - human, machine, software bot,
     conversation archiver, or other, that is bounded by the extent of
!    that entity's [TrustedComputingBase].
  
  4.2.  Conversation
  
--- 239,258 ----
  
     A participant is any entity - human, machine, software bot,
     conversation archiver, or other, that is bounded by the extent of
!    that entity's Trusted Computing Base (TCB).
! 
!    It is important to note that the concept of "participant", defined by
!    their TCB, is a foundation for all other trust.  Developing from the
!    basic definitions of a [TrustedComputingBase] and from the concept of
!    "trust-to-trust" discussed in [RoleOfTrust], a failure of a
!    participant to maintain integrity over their own (trusted)
!    environment should not be considered a failure of an E2ESM that
!    connects it to other participants.
! 
!    Further: it is obviously possible to burden the E2ESM with surfacing
!    potential integrity issues of any given participant; but to require
!    such in this standard would risk harming the privacy of the
!    participant entity.
  
  4.2.  Conversation
  
***************
*** 258,263 ****
--- 275,287 ----
  
     2.  The content starts with the word "Hello"
  
+ 
+ 
+ Muffett                 Expires 11 November 2021                [Page 5]
+ 
+ Internet-Draft                    e2esm                         May 2021
+ 
+ 
     3.  The top bit of the first byte of the content in ASCII encoding,
         is zero
  
***************
*** 275,292 ****
     For transport encryption of content, precise Size PCASM SHOULD NOT be
     observable.
  
  
  
! Muffett                  Expires 8 November 2021                [Page 5]
! 
! Internet-Draft                    e2esm                         May 2021
! 
! 
! 4.3.3.  Descriptive PCASM
! 
!    Descriptive PCASM is data that describes the "content".
! 
!    Descriptive PCASM would include, non-exclusively:
  
     1.  The content contains the substring "ello"
  
--- 299,309 ----
     For transport encryption of content, precise Size PCASM SHOULD NOT be
     observable.
  
+ 4.3.3.  Analytic PCASM
  
+    Analytic PCASM is data that describes the "content".
  
!    Analytic PCASM would include, non-exclusively:
  
     1.  The content contains the substring "ello"
  
***************
*** 314,319 ****
--- 331,343 ----
     constitute PCASM, is an OPTIONAL choice for the E2ESM software, but
     that choice MUST be made apparent to participants.
  
+ 
+ 
+ Muffett                 Expires 11 November 2021                [Page 6]
+ 
+ Internet-Draft                    e2esm                         May 2021
+ 
+ 
  4.3.5.  Non-PCASM
  
     Information which would not be PCASM would include, non-exclusively:
***************
*** 330,343 ****
  
     6.  The content was sent from the following geolocation: ...
  
- 
- 
- 
- Muffett                  Expires 8 November 2021                [Page 6]
- 
- Internet-Draft                    e2esm                         May 2021
- 
- 
     7.  The content was composed using the following platform: ...
  
  4.4.  Backdoor
--- 354,359 ----
***************
*** 366,371 ****
--- 382,399 ----
     "intention or opportunity" - and for all of them an access
     facilitation mechanism is first REQUIRED.
  
+ 
+ 
+ 
+ 
+ 
+ 
+ 
+ Muffett                 Expires 11 November 2021                [Page 7]
+ 
+ Internet-Draft                    e2esm                         May 2021
+ 
+ 
     An access facilitation mechanism is a "door", and is inherently
     [DualUse].  Because the goal of E2ESM is to limit access to PCASM
     exclusively to a defined set of participants, then the intended means
***************
*** 384,399 ****
     already allow for the term to be prefixed with "intentional" or
     "unintentional".
  
- 
- 
- 
- 
- 
- Muffett                  Expires 8 November 2021                [Page 7]
- 
- Internet-Draft                    e2esm                         May 2021
- 
- 
     Thus it seems appropriate to use this term in this context, not least
     because it is also not far removed from the similar and established
     term "side channel".
--- 412,417 ----
***************
*** 424,438 ****
     as a conversation participant or as a non-participant.  (Transparency
     of Participation)
  
     If FooBook decides to represent itself as a non-participant, then it
     MUST NOT have any access to PCASM.  (Integrity of Participation /
     Non-Participation)
  
     If FooBook decides to represent itself as a participant, then it MUST
!    NOT have privileged access to PCASM, for instance via direct database
!    access, but it MAY have "normal" access to PCASM of conversations
!    where it is a participant.  (Integrity of Participation, Equality of
!    Participation)
  
     FooBook MAY retain means to eject reported abusive participants from
     a conversation.  (Decrease in Closure of Participation)
--- 442,465 ----
     as a conversation participant or as a non-participant.  (Transparency
     of Participation)
  
+ 
+ 
+ 
+ Muffett                 Expires 11 November 2021                [Page 8]
+ 
+ Internet-Draft                    e2esm                         May 2021
+ 
+ 
     If FooBook decides to represent itself as a non-participant, then it
     MUST NOT have any access to PCASM.  (Integrity of Participation /
     Non-Participation)
  
     If FooBook decides to represent itself as a participant, then it MUST
!    NOT have exceptional access to PCASM, despite being the provider of
!    the service - for instance via raw database access or network
!    sniffing.  However it MAY participate in E2ESM conversations in a
!    "normal" way, and thereby have "normal" access to intra-conversation
!    PCASM.  (Integrity of Participation, Equality of Participation)
  
     FooBook MAY retain means to eject reported abusive participants from
     a conversation.  (Decrease in Closure of Participation)
***************
*** 443,455 ****
     another, treating it as an ejection, followed by an "illegitimate"
     insertion.  (Increase in Closure of Participation)
  
- 
- 
- Muffett                  Expires 8 November 2021                [Page 8]
- 
- Internet-Draft                    e2esm                         May 2021
- 
- 
     FooBook MUST enable the user to observe and manage the complete state
     of their [TrustedComputingBase] with respect to their FooBook
     messaging services.  (Management and Visibility)
--- 470,475 ----
***************
*** 478,483 ****
--- 498,511 ----
     [I-D.knodel-e2ee-definition]
                Knodel, M., Baker, F., Kolkman, O., Celi, S., and G.
                Grover, "Definition of End-to-end Encryption", Work in
+ 
+ 
+ 
+ Muffett                 Expires 11 November 2021                [Page 9]
+ 
+ Internet-Draft                    e2esm                         May 2021
+ 
+ 
                Progress, Internet-Draft, draft-knodel-e2ee-definition-00,
                22 February 2021, <https://tools.ietf.org/html/draft-
                knodel-e2ee-definition-00>.
***************
*** 499,510 ****
                BlueprintForFreeSpeech, "Ricochet Refresh", 2021,
                <https://www.ricochetrefresh.net>.
  
! 
! 
! Muffett                  Expires 8 November 2021                [Page 9]
! 
! Internet-Draft                    e2esm                         May 2021
! 
  
     [TrustedComputingBase]
                Wikipedia, "Trusted Computing Base", 2021,
--- 527,537 ----
                BlueprintForFreeSpeech, "Ricochet Refresh", 2021,
                <https://www.ricochetrefresh.net>.
  
!    [RoleOfTrust]
!               Clark, D. D. and M. S. Blumenthal, "The End-to-End
!               Argument and Application Design: The Role of Trust", 2011,
!               <https://www.repository.law.indiana.edu/fclj/vol63/
!               iss2/3>.
  
     [TrustedComputingBase]
                Wikipedia, "Trusted Computing Base", 2021,
***************
*** 530,560 ****
  
  
  
! 
! 
! 
! 
! 
! 
! 
! 
! 
! 
! 
! 
! 
! 
! 
! 
! 
! 
! 
! 
! 
! 
! 
! 
! 
! 
! 
! Muffett                  Expires 8 November 2021               [Page 10]
--- 557,560 ----
  
  
  
! Muffett                 Expires 11 November 2021               [Page 10]
